tests/integration_tests/releases.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import logging
+from typing import Optional
+
+from packaging import version
+
+from cloudinit import subp
+from cloudinit.subp import ProcessExecutionError
+from tests.integration_tests import integration_settings
+
+log = logging.getLogger("integration_testing")
+
+
+def get_all_ubuntu_series() -> list:
+    """Use distro-info-data's ubuntu.csv to get a list of Ubuntu series"""
+    out = ""
+    try:
+        out, _err = subp.subp(["ubuntu-distro-info", "-a"])
+    except ProcessExecutionError:
+        log.info(
+            "ubuntu-distro-info (from the distro-info package) must be"
+            " installed to guess Ubuntu os/release"
+        )
+    return out.splitlines()
+
+
+def ubuntu_version_from_series(series) -> str:
+    try:
+        out, _err = subp.subp(
+            ["ubuntu-distro-info", "--release", "--series", series]
+        )
+    except subp.ProcessExecutionError as e:
+        raise ValueError(
+            f"'{series}' is not a recognized Ubuntu release"
+        ) from e
+    return out.rstrip(" LTS")
+
+
+@functools.total_ordering
```

Comment:
Doing comparisons of series is great. How do you envision supporting multiple distribution comparisons for tests that may be applicable to both (debian < bookworm) or (ubuntu < lunar) etc

Reply:
It's slightly more verbose, but something like this should work, right?
```python
# Ubuntu
BIONIC = Release("ubuntu", "bionic", "18.04")
FOCAL = Release("ubuntu", "focal", "20.04")
JAMMY = Release("ubuntu", "jammy", "22.04")
KINETIC = Release("ubuntu", "kinetic", "22.10")
LUNAR = Release("ubuntu", "lunar", "23.04")

# Debian
BULLSEYE = Release("debian", "bullseye", "11")
BOOKWORM = Release("debian", "bookworm", "12")

CURRENT_RELEASE = Release.from_os_image()
IS_UBUNTU = CURRENT_RELEASE.os == "ubuntu"
IS_DEBIAN = CURRENT_RELEASE.os == "debian"
```

Then:
```python
@pytest.mark.skipif(
    (IS_UBUNTU and CURRENT_RELEASE < LUNAR)
    or (IS_DEBIAN and CURRENT_RELEASE < BOOKWORM),
    reason="The turbo encabulator doesn't exist in older versions",
)
```
for more than 2, using `any()` might look nicer.